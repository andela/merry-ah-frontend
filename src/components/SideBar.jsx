import React from 'react';

const SideBar = () => (
  <div className="wrapper">
    <nav id="sidebar">
      <ul className="list-unstyled components">
        <li className="active">
          <a className='side'><i className="fa fa-user"></i> My Page</a>
        </li>
        <li>
          <a className='side'>
              <i className="fa fa-cog"></i> Settings</a>
        </li>
      </ul>
    </nav>
    <div className="clear-fix"></div>
  </div>
);

export default SideBar;
