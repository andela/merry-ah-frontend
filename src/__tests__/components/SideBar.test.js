/* eslint-disable import/no-unresolved */
import 'babel-polyfill';
import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import SideBar from '../../components/SideBar.jsx';
import { props } from '../__mocks__/usersMock';

let wrapper;

beforeEach(() => {
  wrapper = mount(
    <MemoryRouter>
      <SideBar {...props} />
    </MemoryRouter>,
  );
});
describe('SideBar Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
