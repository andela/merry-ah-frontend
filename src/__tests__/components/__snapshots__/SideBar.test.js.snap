// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar Component should match snapshot 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "tdu28q",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "tdu28q",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <SideBar
      auth={
        Object {
          "user": Object {
            "id": 7,
          },
        }
      }
      editProfile={[Function]}
      getUserArts={[Function]}
      getUserProfile={[Function]}
      user={
        Object {
          "isGettingArts": false,
          "userArts": Array [
            Object {
              "Category": Object {
                "categoryName": "Poetry",
              },
              "description": "some descriptions",
              "featuredImg": "image.url",
              "slug": "isuguixu",
              "title": "some title",
            },
          ],
          "userProfile": Object {
            "id": 7,
            "profile": Object {
              "bio": "my bio",
              "firstName": "Juliet",
              "imgURL": "img.URL",
              "lastName": "Ezekwe",
            },
            "userType": "artist",
          },
        }
      }
    >
      <div
        className="wrapper"
      >
        <nav
          id="sidebar"
        >
          <ul
            className="list-unstyled components"
          >
            <li
              className="active"
            >
              <a
                className="side"
              >
                <i
                  className="fa fa-user"
                />
                 My Page
              </a>
            </li>
            <li>
              <a
                className="side"
              >
                <i
                  className="fa fa-cog"
                />
                 Settings
              </a>
            </li>
          </ul>
        </nav>
        <div
          className="clear-fix"
        />
      </div>
    </SideBar>
  </Router>
</MemoryRouter>
`;
